geocoding/generate_pincode.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import math
+import pathlib
+
+import fire
+import googlemaps
+import pandas as pd
+from tqdm import tqdm
+
+logging.basicConfig(
+    filename="pincoding.log",
+    filemode="a",
+    format="%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s",
+    datefmt="%H:%M:%S",
+    level=logging.INFO,
+)
+
+logging.info("Logger Setup Complete")
+
+
+def send_reverese_geocoding_request(lat, lng, gmaps):
+    """Responsible for sending the request to GMaps API to reverser geocode
+    
+    Args:
+        lat (float): Latitude
+        lng (float): Longitude
+        gmaps (Object): gmaps object with declared API key
+    
+    Returns:
+        dict: Response recieved from API
+    """
+    successful_api_hit = True
+    try:
+        reverse_geocode_result = gmaps.reverse_geocode((lat, lng))
+    except:
+        logging.warning(f"Failure in reaching the API - {(lat,lng)}")
+        successful_api_hit = False
+        reverse_geocode_result = None
+    return reverse_geocode_result, successful_api_hit
+
+
+def retrieve_pincodes_from_response(reverse_geocode_result):
+    """This takes the raw response from the API and gathers all the possible pincodes returned by the API
+    
+    Args:
+        reverse_geocode_result (dict): Response from GMaps API
+    
+    Returns:
+        List: List of all the possible pincodes
+    """
+    codes = []
+    result_dicts = reverse_geocode_result[0]["address_components"]
+    for result_dicts_complete in reverse_geocode_result:
+        result_dicts = result_dicts_complete["address_components"]
+        for result_dicts in result_dicts:
+            if "postal_code" in result_dicts["types"]:
+                codes.append(result_dicts["long_name"])
+    return codes
+
+
+def reverse_geocode(csv_sheet, api_key, output_file="AssamPincode.csv"):
+    """Function which uses GMap Geocoding API to find the pincode using given latitude longitude.
+    Saves in the same CSV sheet passed as parameter.
+    
+    Args:
+        csv_sheet (str): CSV sheet that needs to be reverse geocoded for pincodes
+        api_key (str): API Key to access the GMap API
+        output_file (str, optional): Name of the new CSV to be stored as. Defaults to 'AssamNewAddressPincode'.
+    """
+    gmaps = googlemaps.Client(key=api_key)
+    csv_sheet = pathlib.Path.cwd() / csv_sheet
```

Comment:
```suggestion
    csv_sheet = pathlib.Path.cwd() / csv_sheet
```

What can we do to remove the current directory compulsion? Maybe just take the complete path from function input? And then we can resolve it? And we won't have to prepend the `cwd`?

Reply:
Removed the `cwd` compulsion. Works with `Path.resolve()`.